asv_bench/benchmarks/strings.py

```diff
@@ -230,17 +230,24 @@ def time_contains(self, dtype, regex):
 
 class Split:
 
-    params = [True, False]
-    param_names = ["expand"]
+    params = (["str", "string", "arrow_string"], [None, "-", "--"], [True, False])
```

Comment:
I would leave out the `pat` from this benchmark, to reduce the combinatory explosion of cases. Which pattern is being used shouldn't influence the performance of expanding or not. So I would benchmark them separately.

Reply:
Indeed. This is from broken off #41085 where sep determines the path taken, either pyarrow.compute.utf8_split_whitespace, pyarrow.compute.utf8_split_whitespace, pyarrow.compute.split_pattern or object fallback.

so can remove from this PR for now.